### DISCUSSION TITLE
How to read DDR Mode Registers

### DISCUSSION THREAD

**Discussion by tpearman353** [polarfire-soc/polarfire-soc-documentation#123] [Bare metal embedded software]
Hello,

Is it possible to read the mode registers (MRs) of a LPDDR4 memory with the Polarfire SoC? I see how a mode register write is done with mode_register_write() in mss_ddr.c, but cannot figure out how a read is done. It looks like the INIT_MRR_MODE register might be relevant. Also, I thought INIT_MR_ADDR might be used to set which MR to read, but the description from register map documentation seems to indicate this is used for MR writes only.

Thanks,
Taylor

**Comment by mallynch** [comment]
Hi Taylor,

I believe the mode registers can not be read in the PolarFire SoC case. Let me reach out to the relevant person and I will revert early next week.

Regards,
Malachy

**Reply by alee1234567** [reply]
Hey did you ever get the answer to this question? I'm curious myself

**Reply by mallynch** [reply]
Hi,

We have added code to read lpddr4 mode registers after training but before general operation. This is planned to be released in 2022.12 bare metal release.
